api4/config.go

```diff
@@ -107,6 +107,18 @@ func updateConfig(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	LockTeammateValChanged := !*appCfg.TeamSettings.LockTeammateNameDisplay && *cfg.TeamSettings.LockTeammateNameDisplay
+	NameDisplayChanged := *cfg.TeamSettings.LockTeammateNameDisplay && *cfg.TeamSettings.TeammateNameDisplay != *appCfg.TeamSettings.TeammateNameDisplay
+	if LockTeammateValChanged || NameDisplayChanged {
+		preference := model.PreferenceForAll{Category: "display_settings", Name: "name_format", Value: *cfg.TeamSettings.TeammateNameDisplay}
+		err = c.App.UpdatePreferenceForAll(preference)
```

Comment:
if this is either a team or server setting, why are we storing it within the user preferences?
following the above I wonder if it shouldn't end up in the config.json file?

thinking about it a little bit, my guess is that you didn't want to change how it's rendered, so you change it on the user preferences. but if the admin changes the setting and reverts, it'll change how it displays for all the users without option for rollback. Having a separate master setting for the server/team would prevent this, users can have their preference and also be overriden if the admin so desires.

It would also remove the need for that new struct.

Reply:
Fixed. Thanks!